---
title: "Example Data Preprocessing"
author: "Michael Lin"
date: "02/01/2021"
output: html_document
---

# Dependencies

```{r}
library(PhosR)
library(dplyr)
```


# Loading Data

The ESC phosphoproteomic dataset is used.

This dataset contains phosphoproteomic data from mouse liver cells.

Labels are formatted as **gene;site;amino acid sequence;** where **;** is used as delimiters.

```{r}
load('./Phospho_ESC.RData')
head(dat)
```

# Preprocessing

The observation labels (rownames of ESC dataframe), site labels, and sequence information for each observation are stored.

```{r}
ids = rownames(dat)
sites <- sapply(strsplit(ids, ";"), function(x)paste(x[1], x[2], "", sep=";"))
sequences <- sapply(strsplit(ids, ";"), function(x)x[3])
```


# Extracting Kinase Labels from PhosphositePlus

Class labels are retrieved from the PhosphositePlus database using the *PhosR* package.
Known MTOR (mechanistic target of rapamycin) substrates for mouse cells are extracted below.

```{r}
# Dataframe index of known substrates for MTOR
substrate.ids = which(sites %in% PhosR::PhosphoSite.mouse$MTOR)
y = rep(0, length(sites))
y[substrate.ids] = 1
```


# Extracting Sequence Information

The motif score is computed based off the substrates and sequence information found above.

```{r}
seq.score.raw = frequencyScoring(sequences, createFrequencyMat(sequences[substrate.ids]))
```

# Normalisation

The phosphoproteomic data and sequence score are normalised with min-max normalisation.

```{r}
phospho = (dat - min(dat))/(max(dat) - min(dat))
seq.score = (seq.score.raw - min(seq.score.raw))/(max(seq.score.raw) - min(seq.score.raw))
```

# Training Dataset 

The training dataset has columns of the following format 

- site: how each observation will be identified. In this case, it is gene;site;amino acid sequence;.

- phosphoproteomic data 
- sequence score 
- y: class labels 

**Note.** *site* and *y* are specific column names that must be included in the training dataset.

```{r}
training_df = data.frame(site=ids,
                         phospho,
                         score=seq.score,
                         y)
head(training_df)
```

```{r}
write.csv(training_df,
          './Example_ESC_MTOR_train.csv',
          row.names=FALSE)
```


# Test Dataset 

The test dataset has columns of the following format 

- site: how each observation will be identified. In this case, it is gene;site;amino acid sequence;.

- phosphoproteomic data 
- sequence score

```{r}
test_df = data.frame(site=ids,
                     phospho,
                     score=seq.score)
head(test_df)
```

```{r}
write.csv(test_df,
          './Example_ESC_MTOR_test.csv',
          row.names=FALSE)
```

# Raw Dataset 

A *raw dataset* is created here. 
This dataset is of the form 

- site: observations are labelled with the following format **gene;site;amino acid sequence;**.
- phosphoproteomic data: from *dat* variable, this is the unnormalised phosphoproteomic data
- y: binary class labels (0) negative and (1) positive/known substrate

The dataset is used as an example input phosphoproteomic dataset for an automated preprocessing of sequence score and normalisation with a companion R function **preprocessSnapKinTraining** included in this repository in helpers.R.

```{r}
df = data.frame(dat,
                y)
rownames(df) = ids

write.csv(df,
          './Example_ESC_Raw.csv',
          row.names = FALSE)
```


## Example 

The helper function **preprocessSnapKinTraining** can be used as follows.

```{r}
source('../helpers.R')
preprocessed_data = preprocessSnapKin(df)

train_df = preprocessed_data$training
test_df = preprocessed_data$test

# write.csv(train.df,
#             './Example_ESC_MTOR_train.csv',
#             row.names=FALSE)

# write.csv(test_df,
#           './Example_ESC_MTOR_test.csv',
#           row.names=FALSE)
```










