<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Michael Lin" />

<meta name="date" content="2021-02-01" />

<title>SnapKin Workflow in R</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">SnapKin Workflow in R</h1>
<h4 class="author">Michael Lin</h4>
<h4 class="date">02/01/2021</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PhosR)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by &#39;GGally&#39;:
##   method from   
##   +.gg   ggplot2</code></pre>
<div id="load-dataset" class="section level1">
<h1>Load Dataset</h1>
<p>For this vignette, we will be using the ESC dataset.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../Data/Phospho_ESC.RData&#39;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dat)</span></code></pre></div>
<pre><code>##                                                           X0m_1       X0m_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM;  0.01149286  0.22245040
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH; -1.01596157  1.73590472
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.27493793 -0.05160492
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.25122102 -0.13989938
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; -0.43128987 -0.32753327
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -0.06690762 -0.14065903
##                                                           X0m_3      X0m_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM;  0.02662266 -0.2621297
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH; -1.10686482  0.3853579
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.58007363  0.3551768
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.43187532  0.3189899
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.11406336  0.6431960
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -0.10267025  0.3086731
##                                                           X5m_1       X5m_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -1.05492137 -0.87060694
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  3.09955325  3.47771515
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.16254168  0.88071364
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.12796090 -0.11489989
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; -0.30039655  0.81194641
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -0.08932122  0.07018632
##                                                          X5m_3      X5m_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.4533454 -0.7370281
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  3.2261075  3.8766158
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  1.0290659 -0.2076382
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.8506188 -0.2436157
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  1.3115020  0.8011264
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.0124586  0.2152063
##                                                         X15m_1     X15m_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -1.0210040 -1.2751459
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  3.2311183  3.4711549
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.6587576  0.8422238
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.6057319  0.7105632
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.5516552  0.6276972
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.3096968  0.2436706
##                                                          X15m_3      X15m_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.72159530 -0.65504267
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  3.32101816  3.10053193
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  1.04601153  0.80899492
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.87847050  0.96211291
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.28739454 -0.16809349
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.00898755 -0.01174949
##                                                         X30m_1     X30m_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.2925751 -0.4565856
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  2.2672007  2.8734603
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.3969690  0.3632332
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.3799802  0.3410264
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; -0.2733960 -0.1051063
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -1.1498513 -0.6873953
##                                                          X30m_3     X30m_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM;  0.09351926 -0.1363927
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  3.11563622  2.7962478
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.10818145 -2.5887498
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.01543244 -2.6222106
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; -0.41587912 -0.1840961
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -1.44553735 -0.4233837
##                                                          X1h_1      X1h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -1.0711787 -0.3841021
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  1.5929509  1.2568738
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.9389000 -2.4841078
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -1.8277603 -0.2522731
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; -0.2987353  0.2499344
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -0.9900628 -0.8484697
##                                                           X1h_3       X1h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.26576970 -0.30174423
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  1.59094528  1.70100596
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -1.07539197 -1.98495640
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -2.27499083 -1.36569005
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.22823333 -0.03671236
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -0.05574864 -0.16005358
##                                                          X3h_1        X3h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -1.0609622 -0.235895153
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  1.1133695  0.649841524
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.3975190  0.085371496
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.3847132  0.009095504
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; -0.2084984 -0.581075416
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; -0.3103108 -0.400202729
##                                                          X3h_3      X3h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.4453388 -0.1226915
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  1.3649095  0.9754448
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.4390131  0.2735063
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.3234173  0.2375591
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; -0.3497703 -0.6988569
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.2406548 -0.1453142
##                                                          X6h_1      X6h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.3280247 -0.6188259
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH; -0.7307042  1.2605142
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -1.8927884 -0.9759818
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.8830394 -1.2411623
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.8687110  0.1550495
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.8099417  0.7159002
##                                                          X6h_3      X6h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.6748315 -0.6059397
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  0.8557470 -0.4061765
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.8515676 -0.2705879
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -1.0994933 -0.3063163
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.4671856  0.5665934
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.7778156  0.7883770
##                                                          X12h_1     X12h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.61350363 -1.4958218
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH; -1.06864671 -1.6138105
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.08216747 -0.6146327
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.05977753 -0.8029216
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  1.26931778  1.2831638
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.13187931  0.1312871
##                                                         X12h_3      X12h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.7272450 -0.95963164
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  1.1458456  1.10757365
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.5056385 -0.25416788
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.3886508 -0.28981505
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.6916974  1.16209212
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.1342802  0.05636942
##                                                         X24h_1      X24h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.5331760 -1.11383043
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  2.2053362  1.55245061
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.2305976  0.05000953
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.2053023  0.11349053
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  1.1733825  0.48391448
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.7922893  0.16791766
##                                                         X24h_3      X24h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -1.0615216 -0.41540381
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  1.7704581  1.53991175
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.2739945 -0.05435059
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.2683826 -0.09020013
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.5403465  1.38024438
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.6843551  0.84976285
##                                                         X36h_1      X36h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.5074536 -0.76174439
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  2.3812145  2.55460899
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.5416480 -0.34631987
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.5397535 -0.44801078
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.5744185  0.02660799
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.9962716  0.89498881
##                                                         X36h_3      X36h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -1.0471937 -0.74651023
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  2.7102510  1.86639300
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.1354818 -0.75393479
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.1026094 -0.78923732
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.4622890  0.08852148
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  1.0666152  1.16704327
##                                                         X48h_1      X48h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.9940683 -0.24202608
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  2.8670268  2.60278851
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.4253879  0.35707376
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.4044892  0.32757622
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.1726341 -0.05738781
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.6565406  0.12728148
##                                                         X48h_3     X48h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM;  0.1566981 -0.5963384
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  2.4616130  2.6511249
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; -0.1392550  0.4077843
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; -0.0412416  0.3718154
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.3093082  0.3741274
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.4976030  0.5156460
##                                                         X72h_1     X72h_2
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.9757885 -0.5744145
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  2.0188104  2.0245186
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  0.8972663  1.1364410
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  0.8582576  1.4694217
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.8545117  0.6141149
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.5849001  0.4059405
##                                                         X72h_3     X72h_4
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; -0.7394854 -0.9069461
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;  1.9552237  0.2564770
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;  1.2669864  1.1529764
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;  1.1684641  1.1164896
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;  0.4937585  0.4197106
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;  0.5080230  0.3214990</code></pre>
<p>The ESC dataset consists of time-course phosphoproteomic data stored in a matrix <em>dat</em>.</p>
</div>
<div id="assign-labels" class="section level1">
<h1>Assign Labels</h1>
<p>Since the training set for SnapKin requires labelled data, we begin by identifying known substrates for the kinase MTOR. To do this, we refer to the PhosphositePlus database using the PhosR package.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify sites </span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">=</span> <span class="fu">rownames</span>(dat)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>sites <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(ids, <span class="st">&quot;;&quot;</span>), <span class="cf">function</span>(x)<span class="fu">paste</span>(x[<span class="dv">1</span>], x[<span class="dv">2</span>], <span class="st">&quot;&quot;</span>, <span class="at">sep=</span><span class="st">&quot;;&quot;</span>))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Dataframe index of known substrates for MTOR</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>substrate.ids <span class="ot">=</span> <span class="fu">which</span>(sites <span class="sc">%in%</span> PhosR<span class="sc">::</span>PhosphoSite.mouse<span class="sc">$</span>MTOR)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="fu">length</span>(sites))</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>y[substrate.ids] <span class="ot">=</span> <span class="dv">1</span></span></code></pre></div>
</div>
<div id="snapkin-training-and-test-set" class="section level1">
<h1>SnapKin Training and Test Set</h1>
<p>Now that we have binary labels for our dataset, we can construct a dataframe that can be used to train a SnapKin model with the help of some helper functions. To begin, we can construct the training set as a dataframe consisting of the phosphoproteomic data and labels denoted by <em>training_set</em>. Additionally, we may also construct a test set which only contains the phosphoproteomic data denoted by <em>test_set</em>.</p>
<p><strong>Note.</strong> The ids for a site must be of the form <strong>gene;site;amino acid sequence;</strong>. An additional step may be needed to convert your data into the appropriate format and is not included in this vignette.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preprocessing datasets </span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">=</span> <span class="fu">data.frame</span>(dat, y)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(training_set) <span class="ot">=</span> ids </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>test_set <span class="ot">=</span> <span class="fu">data.frame</span>(dat)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(test_set) <span class="ot">=</span> ids</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Import preprocessSnapKin</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;../R/preprocessSnapKin.R&#39;</span>)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div id="snapkin-preprocessing-preprocesssnapkin" class="section level2">
<h2>SnapKin Preprocessing: preprocessSnapKin</h2>
<p>Now that we have our training set (and test set) formatted appropriately, we may use <em>preprocessSnapKin</em> in order to construct appropriate dataframes for SnapKin. Specifically, this function computes motif scores, normalises the dataset, and includes a <em>site</em> column for labelling purposes based off rownames (or set by the user). We illustrate the various scenarios where <em>preprocessSnapKin</em> may be used.</p>
<div id="training-and-test-set" class="section level3">
<h3>Training and Test Set</h3>
<p>In this scenario, we have a labelled training set and a test set (refer to the dataframes constructed <a href="#SnapKin-Training-Set">above</a>). <em>preprocessSnapKin</em> may be used as follows to create a train and test dataframe for SnapKin.</p>
<p><strong>Note.</strong> The sequence score for the test dataset is based off the labelled training set.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute sequence information and returns a list</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>preprocessed_data <span class="ot">=</span> <span class="fu">preprocessSnapKin</span>(<span class="at">training_set=</span>training_set,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">test_set=</span>test_set)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the training and test df</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>train_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>training</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>test_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>test</span></code></pre></div>
<p>Rather than renaming the rownames of your dataframe before using <em>preprocessSnapKin</em>, <em>preprocessSnapKin</em> may also refer to ids directly.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute sequence information and returns a list</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>preprocessed_data <span class="ot">=</span> <span class="fu">preprocessSnapKin</span>(<span class="at">training_set=</span>training_set,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">training_ids=</span>ids,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">test_set=</span>test_set,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">test_ids=</span>ids)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the training and test df</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>train_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>training</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>test_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>test</span></code></pre></div>
</div>
<div id="training-set" class="section level3">
<h3>Training Set</h3>
<p>In some cases, we may just be interested in seeing how our model will predict on the training set based off the training set. <em>preprocessSnapKin</em> can be used to do this as follows</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute sequence information and returns a list</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>preprocessed_data <span class="ot">=</span> <span class="fu">preprocessSnapKin</span>(<span class="at">training_set=</span>training_set)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the training and test df</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>train_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>training</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>test_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>test</span></code></pre></div>
<p>In this case, the test dataframe is the same as the training dataframe except it doesn’t have labels. Similarly, we can refer to ids directly as before</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute sequence information and returns a list</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>preprocessed_data <span class="ot">=</span> <span class="fu">preprocessSnapKin</span>(<span class="at">training_set=</span>training_set,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">training_ids=</span>ids)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the training and test df</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>train_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>training</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>test_df <span class="ot">=</span> preprocessed_data<span class="sc">$</span>test</span></code></pre></div>
</div>
</div>
</div>
<div id="snapkin" class="section level1">
<h1>SnapKin</h1>
<p>Now that our data is ready for SnapKin to use, we can run SnapKin in R with the help of r-reticulate. For the following to work, we assume that you have created the appropriate conda environment as mentioned in the Github repository. Our example uses the <em>SnapKin</em> conda environment, specifically Tensorflow with CPU.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the conda environment to SnapKin</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>reticulate<span class="sc">::</span><span class="fu">use_condaenv</span>(<span class="at">condaenv =</span> <span class="st">&#39;SnapKin&#39;</span>, <span class="at">required=</span><span class="cn">TRUE</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># reticulate::use_condaenv(condaenv = &#39;SnapKin-GPU&#39;, required=TRUE)</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the SnapKin module</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>Snapkin <span class="ot">=</span> reticulate<span class="sc">::</span><span class="fu">import_from_path</span>(<span class="st">&#39;SnapKin&#39;</span>,<span class="at">path =</span> <span class="st">&#39;../Python/&#39;</span>)</span></code></pre></div>
<p>Since SnapKin takes pandas dataframes as inputs, we convert our training and test datasets into pandas dataframes using <em>r_to_py</em>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the train/test dataset to pandas dataframes.</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>snapkin_train <span class="ot">=</span> reticulate<span class="sc">::</span><span class="fu">r_to_py</span>(train_df)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>snapkin_test <span class="ot">=</span> reticulate<span class="sc">::</span><span class="fu">r_to_py</span>(test_df)</span></code></pre></div>
<p>Now that the datasets are in the right format, we can run SnapKin using <strong>snapkin</strong>, which is a function implemented in Python. We can include various arguments as seen below.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>predictions <span class="ot">=</span> Snapkin<span class="sc">$</span><span class="fu">snapkin</span>(<span class="at">df_train=</span>snapkin_train,</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">df_test=</span>snapkin_test,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">verbose=</span><span class="dv">1</span>,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">num_layers=</span><span class="dv">3</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">num_snapshots=</span><span class="dv">10</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">num_epochs=</span><span class="dv">150</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">batch_size=</span><span class="dv">32</span>,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">learning_rate=</span><span class="fl">0.01</span>)</span></code></pre></div>
<p>This function outputs a R dataframe of SnapKin’s predictions for the test set that can be easily used.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(predictions)</span></code></pre></div>
<pre><code>##                                                                                                    site
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM;
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH; 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH;
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK;
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT;
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK;
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT;
##                                                       prediction
## 1700013H16RIK;S124;RSEELRSTSDSCRERSPVELFTDPGEKVEVM; 4.698749e-03
## 1700013H16RIK;S165;ALSAKRQRMETTARQSFEGVEQKMKEVWNSH; 2.758813e-04
## 2310022A10RIK;S141;SSAASRMIANSLNHDSPPHTPTRRSDNSTSK; 4.178271e-03
## 2310022A10RIK;T145;SRMIANSLNHDSPPHTPTRRSDNSTSKISVT; 4.537820e-05
## 2610021A01RIK;S340;REKPQKYKEYRLEITSPSKLVGHPRSFVQQK; 1.186870e-03
## 2610305D13RIK;S131;STPHKTNHRDATLQSSNLKRHKTGTTKQVCT; 2.289464e-05</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
